.github/workflows/fakedroid-odt.yml

```diff
@@ -3,34 +3,75 @@ on:
   pull_request:
   push:
 jobs:
-  fakedroid-odt:
+  fakedroid-odt-channel:
     runs-on: ubuntu-latest
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
 
       - name: Install nix
-        uses: cachix/install-nix-action@v17
-        with:
-          nix_path: nixpkgs=channel:nixos-22.05
-          extra_nix_config: "experimental-features = nix-command"
+        uses: cachix/install-nix-action@v18
 
       - name: Setup cachix
-        uses: cachix/cachix-action@v10
+        uses: cachix/cachix-action@v12
         with:
           name: nix-on-droid
           signingKey: "${{ secrets.CACHIX_SIGNING_KEY }}"
 
-      - name: Initialize fakedroid
-        run: tests/fakedroid.sh echo INIT
+      - name: Build fakedroid script
+        run: |
+          nix \
+            --log-format bar-with-logs \
+            --option keep-going true \
+            --show-trace \
+            build .#fakedroid \
+            --impure \
+            --out-link fakedroid
+
+      - name: Initialize fakedroid for channel setup
+        run: ./fakedroid echo INIT
 
       - name: Run tests
         run: |
-          tests/fakedroid.sh mkdir -p .cache/nix-on-droid-self-test
-          tests/fakedroid.sh touch .cache/nix-on-droid-self-test/confirmation-granted
-          tests/fakedroid.sh nix-on-droid on-device-test
+          ./fakedroid mkdir -p .cache/nix-on-droid-self-test
+          ./fakedroid touch .cache/nix-on-droid-self-test/confirmation-granted
+          ./fakedroid nix-on-droid on-device-test
+
+      - name: Push to cachix
+        if: always() && github.event_name != 'pull_request'
+        run: ./fakedroid nix-shell -p cachix --run 'nix --extra-experimental-features nix-command path-info --all | cachix push nix-on-droid'
+
+
+  fakedroid-flakes:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install nix
+        uses: cachix/install-nix-action@v18
+
+      - name: Setup cachix
+        uses: cachix/cachix-action@v12
+        with:
+          name: nix-on-droid
+          signingKey: "${{ secrets.CACHIX_SIGNING_KEY }}"
+
+      - name: Build fakedroid script
+        run: |
+          nix \
+            --log-format bar-with-logs \
+            --option keep-going true \
+            --show-trace \
+            build .#fakedroid \
+            --impure \
+            --out-link fakedroid
+
+      - name: Initialize fakedroid for flake setup
+        run: USE_FLAKE=1 ./fakedroid echo INIT
 
```

Comment:
Does not run tests. I anticipate it to fail, BTW, because `nix-on-droid on-device-test` refers to a now non-existing channel.

Reply:
That's why I don't run tests there. If the initial setup works, then it is fine. We test flake stuff already in the channel setup fakedroid session

Reply:
Uh-huh, so, channel-using tests aren't supposed to pass.